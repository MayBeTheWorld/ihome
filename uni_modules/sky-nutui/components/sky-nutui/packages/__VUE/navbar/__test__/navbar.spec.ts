import { mount, config } from '@vue/test-utils';
import NavBar from '../index.vue';

import NutIcon from '../../icon/index.vue';

beforeAll(() => {
  config.global.components = {
    NutIcon
  };
});

afterAll(() => {
  config.global.components = {};
});

test('should render left slot correctly', () => {
  const wrapper = mount(NavBar, {
    slots: {
      left: () => 'Custom Left'
    }
  });
  expect(wrapper.find('.nut-navbar__left').html()).toMatchSnapshot();
});

test('should render right slot correctly', () => {
  const wrapper = mount(NavBar, {
    slots: {
      right: () => 'Custom Right'
    }
  });

  expect(wrapper.find('.nut-navbar__right').html()).toMatchSnapshot();
});

test('should render title slot correctly', () => {
  const wrapper = mount(NavBar, {
    slots: {
      title: () => 'Custom Title'
    }
  });

  expect(wrapper.find('.nut-navbar__title').html()).toMatchSnapshot();
});

test('should left-text', () => {
  const wrapper = mount(NavBar, {
    props: {
      leftText: 'left'
    }
  });

  const label = wrapper.find('.nut-navbar__left .nut-navbar__text');
  expect(label.text()).toBe('left');
});

test('should desc', () => {
  const wrapper = mount(NavBar, {
    props: {
      desc: 'desc'
    }
  });

  const label = wrapper.find('.nut-navbar__right .nut-navbar__text');
  expect(label.text()).toBe('desc');
});

test('should render placeholder element when using placeholder prop', async () => {
  const wrapper = mount(NavBar, {
    props: {
      fixed: true,
      placeholder: true
    }
  });
  expect(wrapper.find('.nut-navbar--placeholder').html()).toMatchSnapshot();
});

test('should emit click-left event when clicking left text', () => {
  const wrapper = mount(NavBar, {
    props: {
      leftText: 'left'
    }
  });

  wrapper.find('.nut-navbar__left').trigger('click');
  expect(wrapper.emitted('on-click-back')).toBeTruthy();
});

test('should emit click-right event when clicking right text', () => {
  const wrapper = mount(NavBar, {
    props: {
      desc: 'right'
    }
  });
  wrapper.find('.nut-navbar__right').trigger('click');
  expect(wrapper.emitted('on-click-right')).toBeTruthy();
});

test('should change z-index when using z-index prop', () => {
  const wrapper = mount(NavBar, {
    props: {
      zIndex: 100
    }
  });
  expect((wrapper.element as HTMLElement).style.zIndex).toEqual('100');
});
