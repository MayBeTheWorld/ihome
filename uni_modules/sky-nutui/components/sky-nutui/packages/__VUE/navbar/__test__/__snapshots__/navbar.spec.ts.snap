// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render left slot correctly 1`] = `
"<view class=\\"nut-navbar__left\\"><i class=\\"nutui-iconfont nut-icon nut-icon-left\\" style=\\"color: rgb(151, 151, 151);\\" src=\\"\\"></i>
  <!--v-if-->Custom Left
</view>"
`;

exports[`should render placeholder element when using placeholder prop 1`] = `
"<view class=\\"nut-navbar--placeholder\\">
  <view class=\\"nut-navbar nut-navbar--fixed\\" style=\\"z-index: 10;\\">
    <view class=\\"nut-navbar__left\\"><i class=\\"nutui-iconfont nut-icon nut-icon-left\\" style=\\"color: rgb(151, 151, 151);\\" src=\\"\\"></i>
      <!--v-if-->
    </view>
    <view class=\\"nut-navbar__title\\">
      <!--v-if-->
      <!--v-if-->
    </view>
    <view class=\\"nut-navbar__right\\">
      <!--v-if-->
    </view>
  </view>
</view>"
`;

exports[`should render right slot correctly 1`] = `
"<view class=\\"nut-navbar__right\\">
  <!--v-if-->Custom Right
</view>"
`;

exports[`should render title slot correctly 1`] = `
"<view class=\\"nut-navbar__title\\">
  <!--v-if-->
  <!--v-if-->
</view>"
`;
